File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int maximum = 1;

        vector<int> subLengths(nums.size(), 1);
        subLengths[0] = 1;
        for (int i = 1; i < nums.size(); i++) {
            for (int j = 0; j < i; j++) {
                if (nums[j] < nums[i]) subLengths[i] = max(subLengths[i], subLengths[j] + 1);
                if (subLengths[i] > maximum) maximum = subLengths[i];
            }
        }

        return maximum;
    }
};

File: Data Structures & Algorithms/burst-balloons/submission-1.java
class Solution {
    int[][] dp;
    public int maxCoins(int[] nums) {
        int[] baloons = new int[nums.length + 2];
        for (int i = 0; i < nums.length; i++) {
            baloons[i+1] = nums[i];
        }
        baloons[0] = 1;
        baloons[nums.length + 1] = 1;

        dp = new int[nums.length+2][nums.length+2];
        
        for (int i = nums.length-1; i >= 0; i--) {
            for (int j = i+2; j <= nums.length+1; j++) {
                
                for (int l = i+1; l < j; l++) {
                    dp[i][j] = Math.max(
                        dp[i][j], 
                        baloons[i] * baloons[l] * baloons[j] + dp[i][l] + dp[l][j]
                    );
                }

            }
        }
        return dp[0][nums.length+1];
    }
}
